docs/architecture.md

```diff
@@ -1,188 +1,260 @@
-that owns the policy is triggered inside the `kubewarden-controller`.
-In this reconciliation loop, a ConfigMap with all the polices bounded to 
-the `PolicyServer` is created. Then the a Deployment rollout of the
-interested `policy-server` is started. As a result of that, the new `policy-server`
-instance will be started with the updated configuration.
+<!--TODO:'reconciliation loop', can we show this loop in the diagram
+somehow? ConfigMap, should be in the diagram? In the OCI registry?-->
 
-At start time, `policy-server` reads its configuration and downloads
-all the Kubewarden policies. Policies can be downloaded from remote
-endpoints like HTTP(s) servers and container registries.
+When creating, modifying or deleting a `ClusterAdmissionPolicy`,
+a reconciliation loop activates in `kubewarden-controller`,
+for the `PolicyServer` owning the policy.
+This reconciliation loop creates a ConfigMap with all the polices bound to the `PolicyServer`.
+Then the Deployment rollout of the `policy-server` starts.
+It results in starting the new `policy-server` instance with the updated configuration.
 
-Policies' behaviour can be tuned by users via policy-specific configuration
-parameters. Once all the policies are downloaded, `policy-server` will ensure
-the policy settings provided by the user are valid.
+<!--TODO:Next paragraph. Download all the policies, that the configuration
+specifies, right?-->
 
-`policy-server` performs the validation of policies's settings by
-invoking the `validate_setting` function exposed by each policy.
-This topic is covered more in depth inside
-of [this section](/writing-policies/spec/01-intro-spec.md) of the documentation.
+At start time, the `policy-server` reads its configuration
+and downloads all the Kubewarden policies specified.
+You can download policies from remote HTTP servers and container registries.
 
-`policy-server` will exit with an error if one or more policies received wrong
-configuration parameters from the end user.
+You use policy configuration parameters to tune a policies' behavior.
+After startup and policy download the `policy-server`
+checks the policy settings provided by the user are valid.
 
-If all the policies are properly configured, `policy-server` will spawn a
-pool of worker threads to evaluate incoming requests using the Kubewarden
-policies specified by the user.
+The `policy-server` validates policy settings by invoking
+the `validate_setting` function exposed by each policy.
+There is further documentation in the
+[writing policies section](/writing-policies/spec/01-intro-spec.md)
+of the documentation.
 
-Finally, `policy-server` will start a HTTPS server that listens to incoming
-validation requests. The web server is secured using the TLS key and certificate
-that have been previously created by `kubewarden-controller`.
+<!--TODO:Next paragraph. 'end-user'? How are the configuration parameters
+specified for the `policy-server`-->
```

Comment:
It's possible that the "end-user" (a.k.a. person who deployed the policy) provide a invalid policy settings. Therefore, there is a function called in the policy module to validate the setting provided by the user in the policy definition. If the function returns that the settings are not valid the policy server will not start until the policy is fixed.

Reply:
for more clarification, this is passed with the [`spec.settings`](https://doc.crds.dev/github.com/kubewarden/kubewarden-controller/policies.kubewarden.io/ClusterAdmissionPolicy/v1@v1.7.0-rc3#spec-settings) of the (Cluster)AdmissionPolicy definition. For example [this](https://github.com/kubewarden/kubewarden-controller/blob/main/config/samples/policies_v1_clusteradmissionpolicy.yaml#L17-L21). See also the quickstart on docs.